ext/opentelemetry-ext-requests/tests/test_requests_integration.py

```diff
@@ -183,6 +185,45 @@ def test_distributed_context(self):
         finally:
             propagators.set_global_httptextformat(previous_propagator)
 
+    def test_apply_custom_attributes_on_span(self):
+        resource = resources.Resource.create({})
+        result = self.create_tracer_provider(resource=resource)
+        tracer_provider, exporter = result
+        RequestsInstrumentor().uninstrument()
+
+        def outgoing_http_custom_attributes(span, result: requests.Response):
+            span.set_attribute("http.request.body", result.request.body)
+            span.set_attribute(
+                "http.response.body", result.content.decode("utf-8")
+            )
+
+        RequestsInstrumentor().instrument(
+            tracer_provider=tracer_provider,
+            apply_custom_attributes_on_span=outgoing_http_custom_attributes,
+        )
+
+        result = requests.post(self.URL, data=json.dumps({"hello": "world"}))
+        self.assertEqual(result.text, "World!")
+
+        span_list = exporter.get_finished_spans()
```

Comment:
If you use `self.trace_provider` as I suggested above, this would be `self.memory_exporter....`